libyaml/src/Text/Libyaml.hs

```diff
@@ -547,6 +554,18 @@ toEventRaw e f = allocaBytes eventSize $ \er -> do
                     anchorP
     unless (ret == 1) $ throwIO $ ToEventRawException ret
     f er
+  where
+    implicitColl (EventMappingStart NoTag _ _) = 1
```

Comment:
Are you saying you'd prefer a single render yes / no function in FormatOptions? If so I agree, but:

- The `NoTag` and `UriTag ""` checks are defensive; things break (by silently disappearing from the output) if you let them through so we should apply these regardless of what the render function might do. (Whether they're applied by pattern matches or by converting to string and checking for null, I'm not particularly fussed. Happy to reinstate the latter - it's more generic.)

- I _think_ the Plain / Quote distinction is explicit in the libyaml API because the choice of which style is eventually used is made by libyaml itself (if you choose AnyStyle) and may depend on the length of the text etc. So if you want to express for example "pick the best style but show tags for quoted styles but not plain styles" (*), you couldn't do that reliably with a unified render function without writing some brittle code to second guess libyaml's layout rules.

(*) I haven't the first idea why you'd do that. The libyaml C API seems to be designed with it in mind though so I'm wary of discounting it entirely.

Reply:
I'd be fine with ensuring that `Implicit` is always used for `NoTag` and `UriTag ""`, but otherwise defer to the user function.